pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -320,6 +314,28 @@ func (c *Reconciler) reconcile(ctx context.Context, pr *v1beta1.PipelineRun, get
 	// and may not have had all of the assumed default specified.
 	pr.SetDefaults(contexts.WithUpgradeViaDefaulting(ctx))
 
+	pipelineMeta, pipelineSpec, err := resources.GetPipelineData(ctx, pr, getPipelineFunc)
+	if err != nil {
+		logger.Errorf("Failed to determine Pipeline spec to use for pipelinerun %s: %v", pr.Name, err)
+		pr.Status.MarkFailed(ReasonCouldntGetPipeline,
+			"Error retrieving pipeline for pipelinerun %s/%s: %s",
+			pr.Namespace, pr.Name, err)
+		return controller.NewPermanentError(err)
+	}
+
+	propagateLabelsFromPipelineToPipelineRun(pr, pipelineMeta)
+	propagateAnnotationsFromPipelineToPipelineRun(pr, pipelineMeta)
+
+	// If the pipelinerun is cancelled, cancel tasks and update status
+	if pr.IsCancelled() {
+		err := cancelPipelineRun(ctx, logger, pr, c.PipelineClientSet)
+		if err != nil {
+			logger.Errorf("Failed to cancel pipelinerun %s: %v", pr.Name, err)
+			return controller.NewPermanentError(err)
+		}
+		return nil
+	}
```

Comment:
It seems a bit weird to move this that *late* in the reconciler loop just for being able to put labels on it. Any reason for not getting only the pipeline meta before reconcile and do the "propagateLabels" earlier instead ?

Reply:
makes sense, moved the propagation of labels and annotations earlier in the reconcile loop instead of moving cancellation later